api/v1alpha1/gateway_types.go

```diff
@@ -195,40 +192,80 @@ type GatewayStatus struct {
 
 // ListenerStatus is the status associated with each listener block.
 type ListenerStatus struct {
-	// Errors is a list of reasons why a given Listener Spec is
-	// not valid. Errors will be empty if the Spec is valid.
-	Errors []ListenerError `json:"errors"`
-
-	//
-	Address string `json:"address"`
+	// Address bound on this listener.
+	Address *ListenerAddress `json:"address"`
+	// Conditions describe the current condition of this listener.
```

Comment:
Why expose status for `Address` and not `Port`? For example, what if the requested `Port` is already bound to another listener? How is the status of listener extensions surfaced?

Reply:
That's a great question, since a port is optional, there could be a use case where a port is unspecified in the spec but the controller wants to specify a port that has been assigned in the status. As far as your example of if a port is already bound to another listener or invalid for some other reason, I think that would be best expressed as a condition - one I have not added yet. Does that sound reasonable?

Reply:
@robscott sounds good.